examples/hysteresis_car_control_time_optimal.py

```diff
@@ -0,0 +1,244 @@
+"""Gearbox example with multiple modes."""
+
+import nosnoc
+import casadi as ca
+import numpy as np
+from math import ceil, log
+import matplotlib.pyplot as plt
+
+# Hystheresis parameters
+v1 = 10
+v2 = 15
+
+# Model parameters
+q_goal = 150
+v_goal = 0
+v_max = 30
+u_max = 5
+
+# fuel costs of turbo and nominal
+Pn = 1
+Pt = 2.5
+
+
+def create_options():
+    """Create nosnoc options."""
+    opts = nosnoc.NosnocOpts()
+    opts.print_level = 2
+    # Degree of interpolating polynomial
+    opts.n_s = 3
+    # === MPCC settings ===
+    # upper bound for elastic variables
+    opts.s_elastic_max = 1e1
+    # in penalty methods  1: J = J+(1/p)*J_comp (direct)  , 0 : J = p*J+J_comp (inverse)
+    opts.objective_scaling_direct = 0
+    # === Penalty/Relaxation paraemetr ===
+    # initial smoothing parameter
+    opts.sigma_0 = 1e1
+    # end smoothing parameter
+    opts.sigma_N = 1e-3  # 1e-10
+    # decrease rate
+    opts.homotopy_update_slope = 0.1
+    # number of steps
+    opts.N_homotopy = ceil(abs(
+        log(opts.sigma_N / opts.sigma_0) / log(opts.homotopy_update_slope))) + 1
+    opts.comp_tol = 1e-14
+
+    # IPOPT Settings
+    opts.opts_casadi_nlp['ipopt']['max_iter'] = 500
```

Comment:
I guess, the options should not be used like this now.
See also in the other new examples.